docs/development/codeguide.rst

```diff
@@ -245,32 +244,40 @@ The following example class shows a way to implement this::
     class FloatList:
         def __init__(self, init):
             if isinstance(init, str):
-                init = init.split('‖')
+                init = init.split("‖")
             elif isinstance(init, bytes):
-                init = init.split(b'|')
+                init = init.split(b"|")
             self.x = [float(x) for x in init]
 
         def __repr__(self):
-            # Return unicode object containing no non-ASCII characters
+            # Return a str containing no non-ASCII characters
             return f'<FloatList [{", ".join(str(x) for x in self.x)}]>'
 
         def __bytes__(self):
-            return b'|'.join(bytes(x) for x in self.x)
+            return b"|".join(bytes(x) for x in self.x)
 
         def __str__(self):
             if astropy.conf.unicode_output:
-                return '‖'.join(str(x) for x in self.x)
+                return "‖".join(str(x) for x in self.x)
             else:
-                return self.__bytes__().decode('ascii')
+                return self.__bytes__().decode("ascii")
 
 Additionally, there is a test helper,
-``astropy.test.helper.assert_follows_unicode_guidelines`` to ensure that a
-class follows the Unicode guidelines outlined above.  The following
-example test will test that our example class above is compliant::
+``astropy.test.helper.assert_follows_unicode_guidelines``, to ensure that a
+class follows the Unicode guidelines outlined above regarding
+`~object.__bytes__`, `~object.__str__` and `~object.__repr__`.
+The following example test will check if the ``astropy`` implementation of the
+arcsecond is compliant::
+
+    from astropy import units as u
+    from astropy.tests.helper import assert_follows_unicode_guidelines
 
     def test_unicode_guidelines():
-        from astropy.test.helper import assert_follows_unicode_guidelines
-        assert_follows_unicode_guidelines(FloatList(b'5|4|3|2'), roundtrip=True)
+        assert_follows_unicode_guidelines(u.arcsec, roundtrip=vars(u))
```

Comment:
Thanks for the update! However, why change the example from `FloatList` to `u.arcsec`? A contributor probably has no need to check `u.arcsec` so would be nice if we keep the original example.

Reply:
Rendered doc: https://astropy--18255.org.readthedocs.build/en/18255/development/codeguide.html#unicode-guidelines